### PR TITLE
Add Pod Status when pod is pending by going through pending container

### PR DESCRIPTION
**Pull request by tejal29**
Relates to #172

Add Pod Status when pod is pending by going through pending container.

**Description**

**User facing changes**

n/a

**Next PRs.**

Use this #2928 

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- n/a Mentions any output changes.
- n/a Adds documentation as needed: user docs, YAML reference, CLI reference.
- n/a Adds integration tests if needed.

**Reviewer Notes**

- [x] The code flow looks good. 
- [x] Unit test added.
- [x] User facing changes look good.

**Release Notes**

### CODE REVIEW

File: pkg/skaffold/kubernetes/status_check.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	reason  string
+	message string
+}
+
+func (e *PodErr) Error() string {
+	return fmt.Sprintf("pod in error due to reason: %s, message: %s", e.reason, e.message)
+}
+
+func GetPodDetails(client kubernetes.Interface, ns string, podName string) PodStatus {
+	pi := client.CoreV1().Pods(ns)
```
Comment (balopat) [inline pkg/skaffold/kubernetes/status_check.go]:
```suggestion
	pods := client.CoreV1().Pods(ns)
```

**Comment by tejal29** [comment]
@balopat please take a look again!

File: pkg/skaffold/kubernetes/status_check.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	err   *PodErr
+}
+
+type PodErr struct {
+	reason  string
+	message string
+}
+
+func (e *PodErr) Error() string {
+	return fmt.Sprintf("pod in error due to reason: %s, message: %s", e.reason, e.message)
```
[the remark this answers is not in the document]
Comment (tejal29) [reply pkg/skaffold/kubernetes/status_check.go]: Reason is the error code and message is Message explaining Reason.

[the remark this answers is not in the document]
Comment (tejal29) [reply pkg/skaffold/kubernetes/status_check.go]: an example would be "Reasan: ErrImgPull", Message: "image:tag could not be fetched"
Just `Reason` is not enough.

**Pull request merged**